crates/bdk/src/wallet/utils.rs

```diff
@@ -110,6 +112,16 @@ impl<Pk: MiniscriptKey + ToPublicKey> Satisfier<Pk> for Older {
     }
 }
 
+// The Fisher-Yates shuffling algorithm to randomly order a list of N items: [https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle]
+pub(crate) fn shuffle_slice<T>(list: &mut [T], rng: &mut Box<dyn RngCore>) {
+    let mut current_index = list.len() - 1;
```

Comment:
Can we be sure this won't be called with an empty `list` ?

Reply:
No! I will add that base case